pkg/controller/controller_utils.go

```diff
@@ -298,6 +345,42 @@ func ClusterStatusFromClusterAPI(cluster *clusterapi.Cluster) (*clusteroperator.
 	return &status.ClusterDeploymentStatus, nil
 }
 
+// ClusterDeploymentSpecFromCluster extracts a ClusterDeploymentSpec from the ProviderConfig field of a cluster-api Cluster
+func ClusterDeploymentSpecFromCluster(cluster *clusterapi.Cluster) (*clusteroperator.ClusterDeploymentSpec, error) {
```

Comment:
This is the same as `ClusterSpecFromClusterAPI`, except that this function returns an empty spec when the ProviderConfig is empty. I would argue that a Cluster (associated with a ClusterDeployment) with an empty ProviderConfig is an error. Either way, we should get rid of one of the functions.